DataAccessLayer/QueryParameters/ValidationError.cs

```diff
@@ -74,5 +78,96 @@ public static string GetErrorLink(string path)
 
             return s;
         }
+
+        public static string GetCategory(HttpContext context, out string subCategory, out string mode)
+        {
+            string category;
+
+            string path = RequestLogger.GetPathAndQuerystring(context.Request).ToLowerInvariant();
+
+            if (path.StartsWith("/api/movies?") || path.StartsWith("/api/movies/?"))
+            {
+                category = "Movies";
+                mode = "Query";
+
+                if (path.Contains("year="))
+                {
+                    subCategory = "Year10";
+                }
+                else if (path.Contains("rating="))
+                {
+                    subCategory = "Rating10";
+                }
+                else if (path.Contains("genre="))
+                {
+                    subCategory = "Genre10";
+                }
+                else
+                {
+                    subCategory = "Movies";
+                }
+
+                if (subCategory.EndsWith("10") && path.Contains("pagesize=100"))
+                {
+                    subCategory += "0";
+                }
+            }
+            else if (path.StartsWith("/api/movies/"))
+            {
+                category = "Movies";
+                subCategory = "Movies";
+                mode = "Direct";
+            }
+            else if (path.StartsWith("/api/movies"))
+            {
+                category = "Movies";
+                subCategory = "Movies";
+                mode = "Query";
+            }
+            else if (path.StartsWith("/api/actors?") || path.StartsWith("/api/actors/?"))
+            {
+                category = "Actors";
+                subCategory = "Actors";
+                mode = "Query";
+            }
+            else if (path.StartsWith("/api/actors/"))
```

Comment:
If you move this test to the first actors test then the second test and just look for "/api/actors" and set mode="Query" removing the need for one conditional.